### PR TITLE
feat(search): Tag field options (separator, case sensitivity)

### LINKED ISSUE #3141: Search: Custom delimiter for tags
Add option to change tag delimiter `TAG [SEPARATOR {sep}] [CASESENSITIVE]`

### PR DESCRIPTION
**Pull request by dranikpg**
Fixes #3141

### CODE REVIEW

File: src/core/search/indices.cc [hunk 1]
```diff
@@ -43,12 +47,14 @@ absl::flat_hash_set<std::string> TokenizeWords(std::string_view text) {
...
+                                          char separator) {
   string tmp;
   absl::flat_hash_set<string> tags;
-  for (string_view tag : absl::StrSplit(taglist, ',')) {
+  for (string_view tag : absl::StrSplit(taglist, separator, absl::SkipEmpty())) {
     tmp = absl::StripAsciiWhitespace(tag);
-    absl::AsciiStrToLower(&tmp);
+    if (!case_sensitive)
+      absl::AsciiStrToLower(&tmp);
     tags.insert(std::move(tmp));
```
Comment (dranikpg) [inline src/core/search/indices.cc:58]: > The only two transformations performed are lower-casing (for latin languages only as of now)

Need to check how tags work with unicode to be on the safe side. So far I assumed they're only used with ascii 🤷🏻‍♂️ Can you split by a character (and not by a byte), etc...

**Comment by dranikpg** [comment]
g++ what's that??
image

File: src/core/search/indices.cc [hunk 2]
```diff
@@ -43,12 +47,15 @@ absl::flat_hash_set<std::string> TokenizeWords(std::string_view text) {
 }
 
 // Split taglist, remove duplicates and convert all to lowercase
-absl::flat_hash_set<string> NormalizeTags(string_view taglist) {
+// TODO: introduce ascii support if needed
```
Comment (kostasrim) [inline src/core/search/indices.cc:50]: you mean `unicode` no? 

Also, why don't we just reject the query if the tag is unicode and we uncertain of the behaviour?

**Review by kostasrim — DISMISSED** [review]

File: src/core/search/indices.cc (hunk 2 above)
Comment (dranikpg) [reply src/core/search/indices.cc:50]: Yes, I mean unicode. Would make sense to reject it...

**Comment by dranikpg** [comment]
Added a warning print, propagating errors back from that point on is difficult, because we loose track of the document on that statge 🙄 Hope that'll do, we can look for it on prod

**Review by romange — APPROVED** [review]

**Pull request merged**